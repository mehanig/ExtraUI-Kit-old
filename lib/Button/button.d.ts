import * as React from "react";
export interface IButtonProps {
    onClickHandler?: any;
    text?: string;
    disabled?: any;
    UIType?: string;
}
export interface IButtonState {
    buttonText?: string;
    isDisabled?: boolean;
}
export default class Button extends React.Component<IButtonProps, IButtonState> {
    constructor(props: IButtonProps);
    render(): JSX.Element;
    private onClickHandler(e, obj);
}
