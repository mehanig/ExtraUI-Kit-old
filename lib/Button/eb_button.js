"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var Radium = require("radium");
var React = require("react");
var css_eb_button_1 = require("./css_eb_button");
var EBButton = (function (_super) {
    __extends(EBButton, _super);
    function EBButton(props) {
        var _this = _super.call(this) || this;
        _this.state = {
            onClickHandler: props.onClickHandler,
            buttonText: props.text || "Button",
            isDisabled: props.disabled ? true : false,
        };
        _this.onClickHandler = _this.onClickHandler.bind(_this);
        return _this;
    }
    EBButton.prototype.render = function () {
        var onClick = this.state.isDisabled ? null : this.onClickHandler;
        var stylesArr = (this.props.UIType === "AE") ?
            this.state.isDisabled ? [css_eb_button_1.AEButton, css_eb_button_1.AECssDisabled] : [css_eb_button_1.AEButton, css_eb_button_1.AECssActive]
            :
                this.state.isDisabled ? [css_eb_button_1.CssBase, css_eb_button_1.CssDisabled] : [css_eb_button_1.CssBase, css_eb_button_1.CssActive];
        return (React.createElement("div", null,
            React.createElement(Radium.StyleRoot, null,
                React.createElement("div", { className: "extraui-kit__button-main", style: [stylesArr], onClick: onClick },
                    React.createElement("p", { style: [css_eb_button_1.CssPBase] }, this.state.buttonText)))));
    };
    EBButton.prototype.onClickHandler = function () {
        this.state.onClickHandler();
    };
    return EBButton;
}(React.Component));
exports.EBButton = EBButton;
var Button = Radium(EBButton);
exports.Button = Button;
//# sourceMappingURL=eb_button.js.map