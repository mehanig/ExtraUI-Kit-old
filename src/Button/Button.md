Button examples:

Regular button
```jsx
    <Button/>
```


Disabled button
```jsx
<Button disabled={true} text="disabled"/>
```


Button with `UIType="AE"` is rendered without styling, to be overwritten inside AfterEffects
```jsx
    <Button UIType="AE"/>
```