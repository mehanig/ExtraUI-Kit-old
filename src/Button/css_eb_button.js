"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var CssBase = {
    fontSize: 11,
    display: "flex",
    justifyContent: "center",
    alignContent: "center",
    flexDirection: "column",
    fontFamily: "Helvetica Neue, Helvetica, Arial, sans-serif",
    color: "#fff",
    border: 0,
    width: "140px",
    height: "22px",
    borderRadius: "2px",
    padding: "2px 2px",
    margin: "10px",
    cursor: "pointer",
    outline: "none",
    userSelect: "none",
};
exports.CssBase = CssBase;
var CssActive = {
    "backgroundColor": "#2391FF",
    ":hover": {
        backgroundColor: "#0088FF",
    },
    ":focus": {
        backgroundColor: "#0088FF",
    },
    ":active": {
        backgroundColor: "#005299",
        transform: "translateY(2px)",
    },
};
exports.CssActive = CssActive;
var CssDisabled = {
    backgroundColor: "#393939",
};
exports.CssDisabled = CssDisabled;
var CssPBase = {
    margin: "auto",
};
exports.CssPBase = CssPBase;
