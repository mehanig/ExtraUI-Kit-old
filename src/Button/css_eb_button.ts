import {ICSSProperties} from "../css_types";

const CssBase: ICSSProperties = {
  fontSize: 11,
  display: "flex",
  justifyContent: "center",
  alignContent: "center",
  flexDirection: "column",
  fontFamily: "Helvetica Neue, Helvetica, Arial, sans-serif",
  color: "#fff",
  border: 0,
  width: "140px",
  height: "22px",
  borderRadius: "2px",
  padding: "2px 2px",
  margin: "10px",
  cursor: "pointer",
  outline: "none",
  userSelect: "none",
};

const CssActive: ICSSProperties = {
  "backgroundColor": "#2391FF",
  ":hover": {
    backgroundColor: "#0088FF",
  },
  ":focus": {
    backgroundColor: "#0088FF",
  },
  ":active": {
    backgroundColor: "#005299",
    transform: "translateY(2px)",
  },
};

const CssDisabled: ICSSProperties = {
  backgroundColor: "#393939",
};

const CssPBase: ICSSProperties = {
  margin: "auto",
};

export { CssBase, CssActive, CssDisabled, CssPBase };
