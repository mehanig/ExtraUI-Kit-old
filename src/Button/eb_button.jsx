"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var Radium = require("radium");
var css_eb_button_1 = require("./css_eb_button");
var EBButton = (function (_super) {
    __extends(EBButton, _super);
    function EBButton(props) {
        var _this = _super.call(this) || this;
        _this.state = {
            onClickHandler: props.onClickHandler,
            buttonText: props.text || "Button",
            isDisabled: props.disabled ? true : false,
        };
        _this.onClickHandler = _this.onClickHandler.bind(_this);
        return _this;
    }
    EBButton.prototype.render = function () {
        var onClick = this.state.isDisabled ? null : this.onClickHandler;
        var stylesArr = this.state.isDisabled ? [css_eb_button_1.CssBase, css_eb_button_1.CssDisabled] : [css_eb_button_1.CssBase, css_eb_button_1.CssActive];
        return (<div>
        <Radium.StyleRoot>
          <div style={[stylesArr]} onClick={onClick}><p style={[css_eb_button_1.CssPBase]}>{this.state.buttonText}</p></div>
        </Radium.StyleRoot>
      </div>);
    };
    EBButton.prototype.onClickHandler = function () {
        this.state.onClickHandler();
    };
    return EBButton;
}(React.Component));
exports.EBButton = EBButton;
var Button = Radium(EBButton);
exports.Button = Button;
