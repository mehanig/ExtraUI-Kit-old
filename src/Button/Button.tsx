import * as Radium from "radium";
import * as React from "react";
import {ICSSProperties} from "../css_types";
import {AEButton, AECssActive, AECssDisabled, CssActive, CssBase, CssDisabled, CssPBase} from "./css_eb_button";

export interface IButtonProps {
  /** Callback function to be invoked when button is clicked */
  onClickHandler?: any,
  /** Text to appear on button */
  text?: string,
  /** Specify if button should be non clickable */
  disabled?: any,
  /** Extra parameter to specify if button should be used without styling */
  UIType?: string
}

export interface IButtonState {
  buttonText?: string
  isDisabled?: boolean
}

/**
 * Button component
 */
@Radium
export default class Button extends React.Component<IButtonProps, IButtonState> {
  constructor(props: IButtonProps) {
    super();
    this.state = {
      buttonText: props.text || "Button",
      isDisabled: !!props.disabled,
    };
    this.onClickHandler = this.onClickHandler.bind(this);
  }

  public render() {
    const onClick: any = this.state.isDisabled ? null : this.onClickHandler;
    const stylesArr: [ICSSProperties] = (this.props.UIType === "AE") ?
      this.state.isDisabled ? [AEButton, AECssDisabled] : [AEButton, AECssActive]
    :
      this.state.isDisabled ? [CssBase, CssDisabled] : [CssBase, CssActive];
    return (
      <div>
        <Radium.StyleRoot>
          <div className="extraui-kit__button-main" style={[stylesArr]} onClick={onClick}>
            <p style={[CssPBase]}>{this.state.buttonText}</p></div>
        </Radium.StyleRoot>
      </div>
    );
  }

  private onClickHandler(e: React.MouseEvent<any>, obj: any): void {
    this.props.onClickHandler(e, obj);
  }
}
