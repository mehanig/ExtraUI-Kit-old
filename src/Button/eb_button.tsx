import * as React from "react";
import * as Radium from "radium";
import {ICSSProperties} from "../css_types";
import {CssActive, CssBase, CssDisabled, CssPBase} from "./css_eb_button";

export type StringFunction = () => string;

export interface IButtonProps {
  children?: React.ReactChild,
  className?: string,
  onClickHandler?: any,
  text?: string | StringFunction,
  disabled?: any
}

export interface IButtonState {
  onClickHandler?: any
  buttonText?: string | StringFunction
  isDisabled?: boolean
}

class EBButton extends React.Component<IButtonProps, IButtonState> {
  constructor(props: IButtonProps) {
    super();
    this.state = {
      onClickHandler: props.onClickHandler,
      buttonText: props.text || "Button",
      isDisabled: props.disabled ? true : false,
    };
    this.onClickHandler = this.onClickHandler.bind(this);
  }

  public render() {
    const onClick: any = this.state.isDisabled ? null : this.onClickHandler;
    const stylesArr: [ICSSProperties] = this.state.isDisabled ? [CssBase, CssDisabled] : [CssBase, CssActive];
    return (
      <div>
        <Radium.StyleRoot>
          <div style={[stylesArr]} onClick={onClick}><p style={[CssPBase]}>{this.state.buttonText}</p></div>
        </Radium.StyleRoot>
      </div>
    );
  }

  private onClickHandler(): void {
    this.state.onClickHandler();
  }
}

export { EBButton };
const Button = Radium(EBButton);
export { Button };
