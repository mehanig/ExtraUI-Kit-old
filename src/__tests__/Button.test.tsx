import * as React from "react";
import { create } from "react-test-renderer";
import { Button } from "../Button/eb_button";

describe("Test Empty Button", () => {
  it("renders correctly", () => {
    const tree = create(
      <Button />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
