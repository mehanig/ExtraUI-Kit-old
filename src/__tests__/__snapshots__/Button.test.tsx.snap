// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Empty Button renders correctly 1`] = `
<div>
  <div
    data-radium={true}
  >
    <div
      data-radium={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "alignContent": "center",
          "backgroundColor": "#2391FF",
          "border": 0,
          "borderRadius": "2px",
          "color": "#fff",
          "cursor": "pointer",
          "display": "flex",
          "flexDirection": "column",
          "fontFamily": "Helvetica Neue, Helvetica, Arial, sans-serif",
          "fontSize": 11,
          "height": "22px",
          "justifyContent": "center",
          "margin": "10px",
          "outline": "none",
          "padding": "2px 2px",
          "userSelect": "none",
          "width": "140px",
        }
      }
    >
      <p
        data-radium={true}
        style={
          Object {
            "margin": "auto",
          }
        }
      >
        Button
      </p>
    </div>
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
    />
  </div>
</div>
`;
